StardewArchipelago/Locations/CodeInjections/Modded/SVE/SVEShopInjections.cs

```diff
@@ -177,35 +179,37 @@ private static void MakeBearBarter(ShopMenu shopMenu)
         {
             var stock = shopMenu.itemPriceAndStock;
             var hasKnowledge = _archipelago.HasReceivedItem(BEAR_KNOWLEDGE);
+            var berryItems = _junimoShopGenerator.BerryItems.Keys.ToList();
             foreach (var item in stock)
             {
-            var random = new Random((int)Game1.stats.DaysPlayed + (int)Game1.uniqueIDForThisGame / 2 + item.Key.salePrice());
-            var itemToTrade = _junimoShopGenerator.BerryItems.ElementAt(random.Next(_junimoShopGenerator.BerryItems.Count));
-            if ((item.Key.DisplayName.Contains("Baked Berry Oatmeal") || item.Key.DisplayName.Contains("Flower Cookie")) && _archipelago.SlotData.Chefsanity == Chefsanity.Purchases) // Since these are AP locations, for logic's sake only use seasonal berries
-            {
-                var isOatmeal = item.Key.DisplayName.Contains("Baked Berry Oatmeal");
-                var berry = seasonalBerry[Game1.currentSeason];
-                var berryRate = JunimoShopGenerator.ExchangeRate(isOatmeal ? 12500 : 8750 , berry.salePrice() * (hasKnowledge ? 3 : 1));
-                stock[item.Key] = new int[4]
+                var random = new Random((int)Game1.stats.DaysPlayed + (int)Game1.uniqueIDForThisGame / 2 + item.Key.salePrice());
+                var randomBerryItem = berryItems[random.Next(berryItems.Count)];
+                var randomBerryValue = _junimoShopGenerator.BerryItems[randomBerryItem];
+                if ((item.Key.Name.Contains("Baked Berry Oatmeal") || item.Key.Name.Contains("Flower Cookie")) && _archipelago.SlotData.Chefsanity == Chefsanity.Purchases) // Since these are AP locations, for logic's sake only use seasonal berries
```

Comment:
I'd put the check for chefsanity purchases outside of this loop. Since literally nothing can happen in the loop without it, you could avoid the whole thing when it's not on "Purchases".

Reply:
If I did that none of the other items would be bartered and would only all be barter-able if chefsanity was on.  I put the flag check first so it skips immediately instead.

Reply:
You are correct. I believe I misread the wrongly indented previous version. The current version seems fine. I'd still make a method out of it, but not worth blocking the PR for that